internal/pkg/manifest/workload_applyenv_test.go

```diff
@@ -0,0 +1,1846 @@
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+				svc.Environments["test"].ImageConfig.DependsOn = map[string]string{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.DependsOn = map[string]string{
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+			},
+		},
+		"depends_on not overridden by nil": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.DependsOn = map[string]string{
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.DependsOn = map[string]string{
+					"mockContainer1": "1",
+					"mockContainer2": "2",
+				}
+			},
+		},
+		"port overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Port = aws.Uint16(2)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(2)
+			},
+		},
+		"port explicitly overridden by zero value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Port = aws.Uint16(0)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(0)
+			},
+		},
+		"port not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.Port = aws.Uint16(1)
+			},
+		},
+		"healthcheck overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Retries: aws.Int(3),
+				}
+
+				mockInterval1Minute := 60 * time.Second
+				svc.Environments["test"].ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Interval: &mockInterval1Minute,
+					Retries:  aws.Int(5),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				mockInterval1Minute := 60 * time.Second
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Interval: &mockInterval1Minute,
+					Retries:  aws.Int(5),
+				}
+			},
+		},
+		"healthcheck not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Retries: aws.Int(3),
+				}
+				svc.Environments["test"].ImageConfig.Image = Image{}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.ImageConfig.HealthCheck = &ContainerHealthCheck{
+					Command:     nil,
+					Interval:    nil,
+					Retries:     aws.Int(3),
+					Timeout:     nil,
+					StartPeriod: nil,
+				}
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var inSvc, wantedSvc LoadBalancedWebService
+			inSvc.Environments = map[string]*LoadBalancedWebServiceConfig{
```

Comment:
I think one risk right now is that our tests is only using one workload type: `LoadBalancedWebServiceConfig` which means the other service types' `ApplyEnv` doesn't end up getting covered.

That's another huge refactor that we have to think about but it'd be really good if we can get to a point where the `ApplyEnv` codepath can be shared across services so we don't have to worry about it.

Reply:
I initially have these tests for each of the workload type but they'd be largely repetitive. This is probably also an indication of the need to refactor.

Reply:
I am in the middle of adding a method to the WorkloadManifest interface, which currently only has ApplyEnv...so I'm doubling the amount of repetition! 😬